packages/maskbook/src/plugins/Wallet/services/rpc.ts

```diff
@@ -0,0 +1,102 @@
+import { first } from 'lodash-es'
+import { WalletMessages } from '@masknet/plugin-wallet'
+import type { JsonRpcPayload } from 'web3-core-helpers'
+import { createTransaction } from '../../../database/helpers/openDB'
+import { createWalletDBAccess } from '../database/Wallet.db'
+
+const MAX_UNCONFIRMED_REQUESTS_SIZE = 5
+const MAIN_RECORD_ID = '0'
+
+export async function getUncofirmedRequests() {
+    const t = createTransaction(await createWalletDBAccess(), 'readonly')('UnconfirmedRequestChunk')
```

Comment:
I suggest to store unconfirmed requests in sessionStorage. I think it's unnecessary to store unconfirmed requests across the extension lifetime